pkgs/misc/vim-plugins/update.py

```diff
@@ -488,25 +470,49 @@ def parse_args():
         action="store_true",
         help="Automatically commit updates",
     )
+    parser.add_argument(
+        "--allow-dirty",
+        dest="allow_dirty",
+        action="store_true",
+        help=(
+            "Allow commit to continue even if state is unexpectedly dirty. "
+            "This is only helpful when developing vimPlugins infrastructure."
+        ),
+    )
```

Comment:
This doesn't add much complexity, so I'm fine with keeping it. At the same time it would be nicer to check only the relevant files for "dirt", which would make this unnecessary. If you don't want to implement that, that's fine as well.

Reply:
My thinking here is that there's almost no good reason to have a dirty tree when running this script and the most likely scenario is that you're not ready to run the script. Now that we're automatically committing, people should really create a fresh branch off of master before running. In fact, I'm tempted to be even less permissive and raise an exception if the current branch isn't up to date with `master`.

Reply:
I'm wondering if the added safety checks make this argument inappropriately named. Perhaps `--develop`or `--force` to take off the safety harness.

Reply:
> My thinking here is that there's almost no good reason to have a dirty tree when running this script

Well, almost. You have already identified at least one good reason, which is developing the infrastructure ;)

> and the most likely scenario is that you're not ready to run the script. Now that we're automatically committing, people should really create a fresh branch off of master before running.

Is there a good reason to forbid a dirty tree as long as the files we're touching are clean? The commit will end up the same either way.

> In fact, I'm tempted to be even less permissive and raise an exception if the current branch isn't up to date with `master`.

I wouldn't do that. Just to give one example: Its often not a bad idea to base your work on nixos-unstable or nixpkgs-unstable instead to speed up testing. What would we gain by adding that restriction?